module/modmanager/manager_test.go

```diff
@@ -602,48 +602,3 @@ func TestDebugModule(t *testing.T) {
 		})
 	}
 }
-
-func TestGracefulShutdownWithMalformedModule(t *testing.T) {
-	// This test ensures that module manager's `Add` can be interrupted by a `Close`
-	// call correctly, and no OUE restart goroutines will continue beyond their `inStartup`
-	// check. With our current design, `local_robot.Reconfigure` blocks the main thread,
-	// so the manager will not be `Closed` while a module is being `Add`ed. Future work
-	// (RSDK-4854) may change that though.
-	logger, logs := logging.NewObservedTestLogger(t)
-	// Precompile module to avoid timeout issues when building takes too long.
-	modPath, err := rtestutils.BuildTempModule(t, "module/testmodule")
-	test.That(t, err, test.ShouldBeNil)
-
-	modCfg := config.Module{
-		Name:     "test-module",
-		ExePath:  modPath,
-		LogLevel: "info",
-	}
-
-	// This cannot use t.TempDir() as the path it gives on MacOS exceeds module.MaxSocketAddressLength.
-	parentAddr, err := os.MkdirTemp("", "viam-test-*")
-	test.That(t, err, test.ShouldBeNil)
-	defer os.RemoveAll(parentAddr)
-	parentAddr += "/parent.sock"
-
-	mgr := NewManager(parentAddr, logger, modmanageroptions.Options{UntrustedEnv: false})
-
-	channel := make(chan struct{})
-	go func() {
-		err = mgr.Add(context.Background(), modCfg)
```

Comment:
Somehow the logging changes changed timing such that we hit the race condition writing to this err variable. This turned into a whole discussion about what the test is trying to accomplish. @benjirewis recommended removing the test.